src/components/Modal/Header.tsx

```diff
@@ -0,0 +1,19 @@
+import { Box } from "@saleor/macaw-ui-next";
+import React, { ReactNode } from "react";
+
+import { Close } from "./Close";
+import { Title, TitleProps } from "./Title";
+
+interface HeaderProps extends TitleProps {
+  children: ReactNode;
+  onClose: () => void;
```

Comment:
Suggestion: Wouldn't it be easier to make a provider inside `DashboardModal` and just get `onClose` here from context? From what I see we're passing `onClose` twice everywhere